frontend/packages/topology/src/utils/application-utils.ts

```diff
@@ -180,6 +180,21 @@ export const cleanUpWorkload = async (
     .catch(() => false);
   const buildConfigs = await k8sList(BuildConfigModel, { ns: resource.metadata.namespace });
   const builds = await k8sList(BuildModel, { ns: resource.metadata.namespace });
+  const isServicePresent = await k8sGet(
+    ServiceModel,
+    resource.metadata.name,
+    resource.metadata.namespace,
+  )
+    .then(() => true)
+    .catch(() => false);
```

Comment:
This is a bad pattern to get into. I see that this was a continuation of what was previously implemented. A GET followed by DELETE is still racey.
We should simply delete the resource and ignore 404 errors because the resource doesn't exist.